truffleruby/src/main/ruby/core/truffle/interop.rb

```diff
@@ -63,7 +63,29 @@ def self.enumerable(foreign)
     end
 
     class Foreign
-
+      JAVA_CLASS_CLASS = Foreign.java_class_by_name("java.lang.Class")
+      JAVA_OBJECT_CLASS = java_class_by_name("java.lang.Object")
+      JAVA_FIELD_CLASS = Foreign.java_class_by_name("java.lang.reflect.Field")
+      JAVA_STRING_CLASS = Foreign.java_class_by_name("java.lang.String")
+      
+      CLASS_GET_FIELD = Foreign.get_java_method(
+        JAVA_CLASS_CLASS, "getField", false, JAVA_FIELD_CLASS, JAVA_STRING_CLASS)
+      FIELD_GET = Foreign.get_java_method(
+        JAVA_FIELD_CLASS, "get", false, JAVA_OBJECT_CLASS, JAVA_OBJECT_CLASS)
+      
+      JAVA_INTEGER_CLASS = Foreign.java_class_by_name("java.lang.Integer")
+      JAVA_PRIM_INT_CLASS = Foreign.invoke_java_method(
+        FIELD_GET,
+        Foreign.invoke_java_method(CLASS_GET_FIELD,
+                                   JAVA_INTEGER_CLASS,
+                                   Foreign.to_java_string("TYPE")),
+        nil)
+      OBJECT_HASH_CODE = get_java_method(
+        JAVA_OBJECT_CLASS, "hashCode", false, JAVA_PRIM_INT_CLASS)
```

Comment:
I'm not sure you're understanding me - this all being in Ruby is good, but it being run every time a VM starts and loads the core library isn't. The impact of this isn't just the little time taken to run the Ruby code - the problem it will pull in many more Java classes and require lots of classloading of classes that we wouldn't otherwise need to do something like hello-world. And we can't have un-guarded reflection on core load anyway, due to the SVM.

We only want to run this code if someone starts to use interop. That's `require 'java'` as Benoit says isn't it?